src/dune_engine/process.ml

```diff
@@ -364,9 +364,12 @@ module Exit_status = struct
 
   (* In this module, we don't need the "Error: " prefix given that it is already
      included in the error message from the command. *)
-  let fail paragraphs = raise (User_error.E (User_message.make paragraphs))
+  let fail ~dir paragraphs =
+    match dir with
```

Comment:
I think it's counter intuitive that we don't report the directory just because we inherited it instead of setting it explicitly.

Reply:
I'm not sure I understand, can you elaborate?

Reply:
The directory is

- `None` whenever we it to remain as cwd
- `Some path` whenever we explicitly set it to something.

So it means that errors from running external commands report the directory only when it is explicitly set. This does not  result in predictable error reporting for clients.

Reply:
Ah, yeah, that seems like it could be a problem. Maybe a better place to pass it would be via `Action_exec`?

Reply:
You could also just set dir to cwd if it's None when building the exceptions.